src/web/releases.rs

```diff
@@ -561,11 +561,23 @@ pub fn search_handler(req: &mut Request) -> IronResult<Response> {
             return redirect_to_random_crate(req, &mut conn);
         }
 
-        let (krate, query) = match query.split_once("::") {
-            Some((krate, query)) => (krate.to_string(), format!("?search={query}")),
-            None => (query.clone(), "".to_string()),
+        let mut queries = std::collections::BTreeMap::new();
+
+        let krate = match query.split_once("::") {
+            Some((krate, query)) => {
+                queries.insert("search", query);
+                krate.to_string()
+            }
+            None => query.clone(),
         };
 
+        queries.extend(
```

Comment:
This can be made a little less complex: 

- `if params.contains_key("i-am-feeling-lucky")` above can be changed into `if params.remove("i-am-feeling-lucky").is_some()` 

then this piece can be changed into: 
```rust 
params.remove("query");
queries.extend(params);
```
while being moved into the `if let Ok(matchver) = ...` below